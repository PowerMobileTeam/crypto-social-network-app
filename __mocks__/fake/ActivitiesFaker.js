
function load(count) {

  let activities = [...Array(count)].map((_, i) => {
    const code = 'activityguid' + i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284",
        getAvatarSource: () => {
          return {
            source:'http://thisisaurl'
          }
        }
      },
      message:"Message",
      title:'TITLE',
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"activity",
    }
  });

  return {
    'activities': activities,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function activitiesServiceFaker() {
  return { load: load} 
}

