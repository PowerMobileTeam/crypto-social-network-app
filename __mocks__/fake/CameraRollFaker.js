
function buildPhoto(count) {

  const photos = [];

  return [...Array(count)].map((_, i) => {
    const code = Math.random().toString(16).substring(2).toUpperCase();
    return {
      'node': {
        'timestamp': Date.now(),
        'group_name': 'Camera Roll',
        'type': 'ALAssetTypePhoto',
        'location': {
          'speed': 2.0533344256922823,
          'latitude': 63.5314,
          'heading': 0,
          'longitude': -19.5112,
          'altitude': 0
        },
        'image': {
          'playableDuration': 0,
          'isStored': true,
          'filename': 'IMG_000'+i+'.JPG',
          'width': 3000,
          'height': 2002,
          'uri': 'assets-library://asset/asset.JPG?id=ED7AC36B-A150-4C38-BB8C-'+code+'&ext=JPG'
        }
      }
    }
  });
}

/**
 * getPhotos data faker
 * @param {integer} count
 */
export function getPhotosFaker(count = 3) {
  const photos = buildPhoto(count);
  return {
    'edges': photos,
    'page_info': {
      'has_next_page': false,
      'start_cursor': photos[0].node.image.uri,
      'end_cursor': photos[count-1].node.image.uri
    }
  };
}

