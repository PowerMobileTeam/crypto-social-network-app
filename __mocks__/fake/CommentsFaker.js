
function load(count) {

  let comments = [...Array(count)].map((_, i) => {
    let code = 'guid' +i;
    return {
      attachment_guid: false,
      blurb: false,
      container_guid:code,
      custom_data:false,
      custom_type:false,
      description:"Congratulations! ",
      edited:"",
      guid:code,
      mature:false,
      ownerObj:{
        guid: "824853017709780997", 
        type: "user", 
        subtype: false, 
        time_created: "1522036284"
      },
      owner_guid:"824853017709780997",
      parent_guid:"838106762591510528",
      perma_url:false,
      thumbnail_src:false,
      type:"comment",
    }
  });

  return {
    'comments': comments,
    'load-next': 'aaaaaa',
    'load-previous': 'aaaaaa'
  }
  
}

/**
 * comment service data faker
 * @param {integer} count
 */
export function commentsServiceFaker() {
  return { load: load} 
}

