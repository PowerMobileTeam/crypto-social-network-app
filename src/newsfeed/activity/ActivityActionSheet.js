import React, {
  Component
} from 'react';

import {
  Text,
  Image,
  View,
  ActivityIndicator,
  Button,
  StyleSheet,
  Modal,
  Alert,

} from 'react-native';

import {
  observer,
  inject
} from 'mobx-react/native'

import shareService from '../../share/ShareService';
import { toggleUserBlock } from '../NewsfeedService';
import Icon from 'react-native-vector-icons/MaterialIcons';
import ActionSheet from 'react-native-actionsheet';
import { MINDS_URI } from '../../config/Config';
/**
 * Activity Actions
 */
const title = 'Actions';

@inject("user")
@inject("newsfeed", "navigatorStore")
export default class ActivityActions extends Component {

  constructor(props) {
    super(props)
    this.state = {
      selected: '',
      reportModalVisible: false,
      userBlocked: false,
      options: this.getOptions(),
    }

    this.handleSelection = this.handleSelection.bind(this);
  }

  showActionSheet() {
    this.state = {
      options: this.getOptions(),
    }
    this.ActionSheet.show();
  }

  handleSelection(i) {
    this.makeAction(this.state.options[i]);
  }

  getOptions() {
    let options = [ 'Cancel' ];
    if (this.props.user.me.guid == this.props.entity.ownerObj.guid) {
      options.push( 'Edit' );

      options.push( 'Delete' );

      if (!this.props.entity.mature) {
        options.push( 'Set explicit' );
      } else {
        options.push( 'Remove explicit' );
      }

    } else {

      if (this.props.user.isAdmin()) {
        options.push( 'Delete' );

        if (!this.props.entity.mature) {
          options.push( 'Set explicit' );
        } else {
          options.push( 'Remove explicit' );
        }
      }

      if (this.state && this.state.userBlocked) {
        options.push( 'Unblock user' );
      } else {
        options.push( 'Block user' );
      }

      options.push( 'Report' );
    }

    options.push( 'Share' );

    if (!this.props.entity['is:muted']) {
      options.push( 'Mute notifications' );
    } else {
      options.push( 'Unmute notifications' );
    }


    return options;

  }

  deleteEntity() {
    this.props.newsfeed.list.deleteEntity(this.props.entity.guid).then( (result) => {
      this.setState({
        options: this.getOptions(),
      });

      Alert.alert(
        'Success',
        'Entity removed succesfully',
        [
          {text: 'OK', onPress: () => {}},
        ],
        { cancelable: false }
      )

      if(this.props.navigatorStore.currentScreen == 'Activity' ){
        this.props.navigation.goBack();
      }
    });
  }

  makeAction(option) {
    switch (option) {
      case 'Edit':
        this.props.toggleEdit(true);
        break;
      case 'Delete':
        Alert.alert(
          'Delete',
          "Are you sure? There's no UNDO",
          [
            {text: 'Cancel', onPress: () => console.log('Cancel Pressed'), style: 'cancel'},
            {text: 'OK', onPress: () => this.deleteEntity()},
          ],
          { cancelable: false }
        )
        break;
      case 'Set explicit':
        this.props.newsfeed.list.newsfeedToggleExplicit(this.props.entity.guid).then( (result) => {
          this.setState({
            options: this.getOptions(),
          });
        });
        break;
      case 'Remove explicit':
        this.props.newsfeed.list.newsfeedToggleExplicit(this.props.entity.guid).then( (result) => {
          this.setState({
            options: this.getOptions(),
          });
        });
        break;
      case 'Block user':
        toggleUserBlock(this.props.entity.ownerObj.guid, !this.state.userBlocked).then( (result) => {
          this.setState({
            userBlocked:true,
            options: this.getOptions(),
          });
        });
        break;
      case 'Unblock user':
        toggleUserBlock(this.props.entity.ownerObj.guid, !this.state.userBlocked).then( (result) => {
          this.setState({
            userBlocked:false,
            options: this.getOptions(),
          });
        });
        break;
      case 'Mute notifications':
        this.props.newsfeed.list.newsfeedToggleMute(this.props.entity.guid).then( (result) => {
          this.setState({
            options: this.getOptions(),
          });
        });
        break;
      case 'Unmute notifications':
        this.props.newsfeed.list.newsfeedToggleMute(this.props.entity.guid).then( (result) => {
          this.setState({
            options: this.getOptions(),
          });
        });
        break;
      case 'Monetize':
        this.props.newsfeed.list.toggleMonetization(this.props.entity.guid).then( (result) => {
          this.setState({
            options: this.getOptions(),
          });
        });
        break;
      case 'Un-monetize':
        this.props.newsfeed.list.toggleMonetization(this.props.entity.guid).then( (result) => {
          this.setState({
            options: this.getOptions(),
          });
        });
        break;
      case 'Share':
        shareService.share(this.props.entity.text, MINDS_URI + 'newsfeed/' + this.props.entity.guid);
        break;
      case 'Report':
        this.props.navigation.navigate('Report', { entity: this.props.entity });
        break;
    }


  }

  /**
   * Close report modal
   */
  closeReport = () => {
    this.setState({ reportModalVisible: false });
  }

  /**
   * Render Header
   */
  render() {


    return (
      <View style={styles.wrapper}>
        <Icon
          name="more-vert"
          onPress={() => this.showActionSheet()}
          size={26}
          style={styles.icon}
          />
        <ActionSheet
          ref={o => this.ActionSheet = o}
          title={title}
          options={this.getOptions()}
          onPress={this.handleSelection}
          cancelButtonIndex={0}
        />
      </View>
    )
  }
}

const styles = StyleSheet.create({
  wrapper: {
    flex:1,
    alignSelf: 'center'
  },
  icon: {
    color: '#888',
  },
  iconclose: {
    flex:1,
  },
  modal: {
    flex: 1,
    paddingTop: 4,
  },
  modalContainer: {
    alignItems: 'center',
    backgroundColor: '#ede3f2',
  },
  modalHeader: {
    padding: 5
  }
});