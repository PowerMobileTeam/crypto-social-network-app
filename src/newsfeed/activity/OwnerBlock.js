import React, {
  PureComponent
} from 'react';

import {
  NavigationActions
} from 'react-navigation';

import {
  Text,
  StyleSheet,
  TouchableOpacity,
  Image,
  View
} from 'react-native';

import {
  MINDS_CDN_URI
} from '../../config/Config';

/**
 * Owner Block Component
 */
export default class OwnerBlock extends PureComponent {

  /**
   * Navigate To channel
   */
  _navToChannel = () => {
    // only active if receive the navigation property
    if (this.props.navigation) {
      this.props.navigation.navigate('Channel', { guid:this.props.entity.ownerObj.guid, entity: this.props.entity.ownerObj });
    }
  }

  /**
   * Navigate To group
   */
 _navToGroup = () => {
    if (this.props.navigation) {
      this.props.navigation.navigate('GroupView', { group: this.props.entity.containerObj });
    }
  };

  get group() {
    if(!this.props.entity.containerObj)
      return null;

    return (
      <TouchableOpacity onPress={this._navToGroup} style={styles.groupContainer}>
        <Text style={styles.groupName} lineBreakMode='tail' numberOfLines={1}>
          > { this.props.entity.containerObj.name }
        </Text>
      </TouchableOpacity>
    );
  }

  /**
   * Render
   */
  render() {
    const channel = this.props.entity.ownerObj;
    const rightToolbar = this.props.rightToolbar||null;

    const avatarSrc = channel.getAvatarSource();

    return (
      <View style={styles.container}>
        <TouchableOpacity onPress={this._navToChannel}>
          <Image source={avatarSrc} style={styles.avatar}/>
        </TouchableOpacity>
        <View style={styles.body}>
          <View style={styles.nameContainer}>
            <TouchableOpacity onPress={this._navToChannel}>
              <Text style={styles.username}>
                { channel.username }
              </Text>
            </TouchableOpacity>
            { this.group }
          </View>
          {this.props.children}
        </View>
        {rightToolbar}
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    display: 'flex',
    padding: 8,
    alignItems: 'center',
    flexDirection: 'row',
  },
  avatar: {
    height: 46,
    width: 46,
    borderRadius: 23,
    borderWidth: 0.5,
    borderColor: '#EEE',
  },
  body: {
    marginLeft: 8,
    paddingRight: 36,
    flexWrap: 'wrap',
    flex: 1,
  },
  nameContainer: {
    flexDirection: 'row',
  },
  username: {
    fontWeight: 'bold',
    fontFamily: 'Roboto',
    color: '#444',
  },
  groupContainer: {
    marginLeft: 4,
    flex: 1,
  },
  groupName: {
    fontWeight: 'bold',
    fontFamily: 'Roboto',
    color: '#888',
  }
});
