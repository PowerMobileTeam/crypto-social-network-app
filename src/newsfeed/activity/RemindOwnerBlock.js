import React, {
  PureComponent
} from 'react';

import {
  Text,
  StyleSheet,
  Image,
  TouchableOpacity,
  View
} from 'react-native';

import { Icon } from 'react-native-elements';

import {
  NavigationActions
} from 'react-navigation';

import {
  MINDS_CDN_URI
} from '../../config/Config';

import {Â CommonStyle } from '../../styles/Common';

export default class RemindOwnerBlock extends PureComponent {


  /**
   * Navigate To channel
   */
  _navToChannel = () => {
    // only active if NewsfeedList receive the navigation property
    if (this.props.navigation) {
      this.props.navigation.navigate('Channel', { guid:this.props.entity.ownerObj.guid, entity: this.props.entity.ownerObj });
    }
  }

  render() {
    const entity = this.props.entity.ownerObj;
    const avatarSrc = entity.getAvatarSource();
    return (
      <View style={styles.container}>
        <Icon color='rgb(70, 144, 214)' name='repeat' size={16} style={styles.icon}/>
        <TouchableOpacity onPress={this._navToChannel}>
          <Image source={avatarSrc} style={styles.avatar}/>
        </TouchableOpacity>
        <View style={styles.body}>
          <TouchableOpacity onPress={this._navToChannel} style={[CommonStyle.flexContainer, CommonStyle.columnAlignStart]}>
            <Text style={styles.username}>
              { entity.username }
            </Text>
            { this.props.entity.boosted &&
              <View style={styles.boostTagContainer}>
                <View style={styles.boostTagColumn}>
                  <Icon name="md-trending-up" type='ionicon' size={13} iconStyle={styles.boostTagIcon} />
                </View>
                <View style={styles.boostTagColumn}>
                  <Text style={styles.boostTagLabel}>BOOSTED</Text>
                </View>
              </View>
            }
          </TouchableOpacity>
        </View>
      </View>
    );
  }


}

const styles = StyleSheet.create({
  boostTagContainer: {
    flexDirection: 'row',
    flex:1,
    paddingRight: 32
  },
  boostTagColumn: {
    flexDirection: 'column',
    justifyContent: 'center',
  },
  boostTagIcon: {
    color: '#aaa',
  },
  boostTagLabel: {
    color: '#aaa',
    fontWeight: '800',
    marginLeft: 2,
    fontSize: 10,
  },
  container: {
    display: 'flex',
    flex: 1,
    paddingLeft: 8,
    paddingTop: 8,
    alignItems: 'center',
    flexDirection: 'row',
  },
  icon: {
    marginLeft: 8,
    marginRight: 8,
  },
  avatar: {
    marginLeft: 8,
    height: 24,
    width: 24,
    borderRadius: 12,
    borderWidth: 0.5,
    borderColor: '#EEE',
  },
  body: {
    marginLeft: 8,
    flex:1
  },
  username: {
    fontWeight: 'bold',
    fontFamily: 'Roboto',
    color: '#444',
    fontSize: 13,
  },
});
