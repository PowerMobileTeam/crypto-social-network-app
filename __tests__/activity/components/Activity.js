import 'react-native';
import React from 'react';
import { Text, TouchableOpacity } from "react-native";
import { shallow } from 'enzyme';


import ActionSheet from 'react-native-actionsheet';
import * as Progress from 'react-native-progress';

import Activity from '../../../src/newsfeed/activity/Activity';

import { commentsServiceFaker } from '../../../__mocks__/fake/CommentsFaker';
import { activitiesServiceFaker } from '../../../__mocks__/fake/ActivitiesFaker';

import renderer from 'react-test-renderer';
import ExplicitText from '../../../src/common/components/explicit/ExplicitText';
import OwnerBlock from '../../../src/newsfeed/activity/OwnerBlock';
import Actions from '../../../src/newsfeed/activity/Actions';
import ActivityEditor from '../../../src/newsfeed/activity/ActivityEditor';
import RemindOwnerBlock from '../../../src/newsfeed/activity/RemindOwnerBlock';
import ActivityActionSheet from '../../../src/newsfeed/activity/ActivityActionSheet';
import ActivityMetrics from '../../../src/newsfeed/activity/metrics/ActivityMetrics';

import AutoHeightFastImage from '../../../src/common/components/AutoHeightFastImage';

import formatDate from '../../../src/common/helpers/date';
import domain from '../../../src/common/helpers/domain';
import MediaView from '../../../src/common/components/MediaView';

import Lock from '../../../src/wire/lock/Lock';

jest.mock('../../../src/common/components/explicit/ExplicitText', () => 'ExplicitText');
jest.mock('../../../src/newsfeed/activity/OwnerBlock', () => 'OwnerBlock');
jest.mock('../../../src/newsfeed/activity/Actions', () => 'Actions');
jest.mock('../../../src/newsfeed/activity/ActivityEditor', () => 'ActivityEditor');
jest.mock('../../../src/newsfeed/activity/RemindOwnerBlock', () => 'RemindOwnerBlock');
jest.mock('../../../src/newsfeed/activity/ActivityActionSheet', () => 'ActivityActionSheet');
jest.mock('../../../src/newsfeed/activity/metrics/ActivityMetrics', () => 'ActivityMetrics');

describe('Activity component', () => {

  let user, comments, entity, screen;
  beforeEach(() => {

    const navigation = { navigate: jest.fn() };
    let activityResponse = activitiesServiceFaker().load(1);
    screen = shallow(
      <Activity entity={activityResponse.activities[0]} navigation={navigation} />
    );

    jest.runAllTimers();
  });

  it('renders correctly', async () => {
    screen.update();
    expect(screen).toMatchSnapshot();
  });


  it('should have a Media View', async () => {
    screen.update();

    expect(screen.find(MediaView)).toHaveLength(1);
  });

  it('should have the expectedComponents', async () => {
    screen.update();

    expect(screen.find(MediaView)).toHaveLength(1);
    expect(screen.find(Actions)).toHaveLength(1);
    expect(screen.find(OwnerBlock)).toHaveLength(1);
    expect(screen.find(ExplicitText)).toHaveLength(1);
    expect(screen.find(ActivityMetrics)).toHaveLength(1);
    expect(screen.find(TouchableOpacity)).toHaveLength(1);
  });


  it('should navToActivity on press', () => {
    screen.update()
    let instance = screen.instance();
    const spy = jest.spyOn(instance, 'navToActivity');
    const render = screen.dive();
    render.find('TouchableOpacity').forEach(child => {
      child.simulate('press');
    });
    jest.runAllTimers();

    expect(spy).toHaveBeenCalled();
  });


});
