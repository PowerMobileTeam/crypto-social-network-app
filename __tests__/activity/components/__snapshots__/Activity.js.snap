// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Activity component renders correctly 1`] = `
<Component
  style={
    Object {
      "borderBottomColor": "#EEE",
      "borderBottomWidth": 0.5,
      "overflow": "visible",
    }
  }
>
  <OwnerBlock
    entity={
      Object {
        "attachment_guid": false,
        "blurb": false,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "guid": "activityguid0",
        "mature": false,
        "message": "Message",
        "ownerObj": Object {
          "getAvatarSource": [Function],
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "perma_url": false,
        "thumbnail_src": false,
        "title": "TITLE",
        "type": "activity",
      }
    }
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    rightToolbar={
      <Component
        style={
          Object {
            "alignSelf": "flex-end",
            "position": "absolute",
            "right": 10,
            "top": 20,
          }
        }
      >
        <ActivityActionSheet
          entity={
            Object {
              "attachment_guid": false,
              "blurb": false,
              "container_guid": "activityguid0",
              "custom_data": false,
              "custom_type": false,
              "description": "Congratulations! ",
              "edited": "",
              "guid": "activityguid0",
              "mature": false,
              "message": "Message",
              "ownerObj": Object {
                "getAvatarSource": [Function],
                "guid": "824853017709780997",
                "subtype": false,
                "time_created": "1522036284",
                "type": "user",
              },
              "owner_guid": "824853017709780997",
              "parent_guid": "838106762591510528",
              "perma_url": false,
              "thumbnail_src": false,
              "title": "TITLE",
              "type": "activity",
            }
          }
          navigation={
            Object {
              "navigate": [MockFunction],
            }
          }
          newsfeed={undefined}
          toggleEdit={[Function]}
        />
      </Component>
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#888",
            "fontSize": 11,
          }
        }
      >
        Invalid date
      </Text>
      <ActivityMetrics
        entity={
          Object {
            "attachment_guid": false,
            "blurb": false,
            "container_guid": "activityguid0",
            "custom_data": false,
            "custom_type": false,
            "description": "Congratulations! ",
            "edited": "",
            "guid": "activityguid0",
            "mature": false,
            "message": "Message",
            "ownerObj": Object {
              "getAvatarSource": [Function],
              "guid": "824853017709780997",
              "subtype": false,
              "time_created": "1522036284",
              "type": "user",
            },
            "owner_guid": "824853017709780997",
            "parent_guid": "838106762591510528",
            "perma_url": false,
            "thumbnail_src": false,
            "title": "TITLE",
            "type": "activity",
          }
        }
      />
    </TouchableOpacity>
  </OwnerBlock>
  <Component
    style={
      Object {
        "padding": 8,
      }
    }
  >
    <ExplicitText
      entity={
        Object {
          "attachment_guid": false,
          "blurb": false,
          "container_guid": "activityguid0",
          "custom_data": false,
          "custom_type": false,
          "description": "Congratulations! ",
          "edited": "",
          "guid": "activityguid0",
          "mature": false,
          "message": "Message",
          "ownerObj": Object {
            "getAvatarSource": [Function],
            "guid": "824853017709780997",
            "subtype": false,
            "time_created": "1522036284",
            "type": "user",
          },
          "owner_guid": "824853017709780997",
          "parent_guid": "838106762591510528",
          "perma_url": false,
          "thumbnail_src": false,
          "title": "TITLE",
          "type": "activity",
        }
      }
      navigation={
        Object {
          "navigate": [MockFunction],
        }
      }
      style={
        Object {
          "fontFamily": "Roboto",
        }
      }
    />
  </Component>
  <MediaView
    entity={
      Object {
        "attachment_guid": false,
        "blurb": false,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "guid": "activityguid0",
        "mature": false,
        "message": "Message",
        "ownerObj": Object {
          "getAvatarSource": [Function],
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "perma_url": false,
        "thumbnail_src": false,
        "title": "TITLE",
        "type": "activity",
      }
    }
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
    style={Object {}}
    width={750}
  />
  <Actions
    entity={
      Object {
        "attachment_guid": false,
        "blurb": false,
        "container_guid": "activityguid0",
        "custom_data": false,
        "custom_type": false,
        "description": "Congratulations! ",
        "edited": "",
        "guid": "activityguid0",
        "mature": false,
        "message": "Message",
        "ownerObj": Object {
          "getAvatarSource": [Function],
          "guid": "824853017709780997",
          "subtype": false,
          "time_created": "1522036284",
          "type": "user",
        },
        "owner_guid": "824853017709780997",
        "parent_guid": "838106762591510528",
        "perma_url": false,
        "thumbnail_src": false,
        "title": "TITLE",
        "type": "activity",
      }
    }
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />
</Component>
`;
